medusa/databases/main_db.py

```diff
@@ -257,16 +256,29 @@ def fix_subtitles_codes(self):
                                    [','.join(langs), datetime.datetime.now().strftime(dateTimeFormat), sql_result[b'episode_id']])
 
     def fix_show_nfo_lang(self):
-        self.connection.action("UPDATE tv_shows SET lang = '' WHERE lang = 0 or lang = '0'")
+        self.connection.action("UPDATE tv_shows SET lang = '' WHERE lang = 0 OR lang = '0';")
+
+    def fix_status_qualities(self):
+        """
+        Check for a status bigger than 12, and translate to a status + quality, as these are old composite statuses.
 
+        This can be removed when all code that creates composite statuses has been migrated.
+        Until then this can be used to keep the DB sane.
+        """
+        log.info(u'Convert composite statuses in tv_episodes to status + quality.')
+        sql_results = self.connection.select("SELECT status FROM tv_episodes where status > 12 GROUP BY status;")
+        for status in sql_results:
+            log.info(u'Split composite status into status and quality for %s', status[b'status'])
+            split = common.Quality.split_composite_status(status[b'status'])
```

Comment:
I believe this excludes "simple" statuses from the SQL query, as up to 12 they are not composite (status + quality): https://github.com/pymedusa/Medusa/blob/master/medusa/common.py#L104-L115. The old quality values can't be "simple" in the old status column IIRC, they can only be composite and that would make the > 12.

Reply:
Let's say I have `status = 3204` (which is **Downloaded 720p WEB-DL** on `develop`)
This will extract it, and set the values in database to:
```py
status = 4  # Downloaded
quality = 32  # 1080p HDTV - on this branch
```

What I'm saying is, the value should be shifted `val >> 1`, just like in the migration code at the bottom of this file.
Needs to be:
```py
status = 4  # Downloaded
quality = 64 # 720p WEB-DL - on this branch
```

----

And adding to that, see what the next "sanity check" function does:
```py
    def update_status_unknown(self):
        """Changes any `UNKNOWN` quality to 1."""
        log.info(u'Update status UNKONWN from tv_episodes')
        self.connection.action("UPDATE tv_episodes SET quality = 1 WHERE quality = 65536;")
```

If you didn't shift the quality values, you won't have `quality = 65536` to begin with.